soh/include/functions.h

```diff
@@ -76,7 +76,7 @@ OSPiHandle* osDriveRomInit(void);
 void StackCheck_Init(StackEntry* entry, void* stackTop, void* stackBottom, u32 initValue, s32 minSpace,
                      const char* name);
 void StackCheck_Cleanup(StackEntry* entry);
-StackStatus StackCheck_GetState(StackEntry* entry);
+s32 StackCheck_GetState(StackEntry* entry);
```

Comment:
why would we want to use an s32 instead of an enum type here? is this a common pattern in this PR?

Reply:
> I want to assume that maybe it was wrong to have been using that enum?

curious as to why it would be wrong, i'd much prefer things to use named types like `StackStatus` than `u32`s, but maybe i'm missing something

Reply:
i'd prefer to follow the opposite rule of "always use types that describe what's being returned", but if that'll cause merge problems when pulling in decomp stuff i guess i can understand following their pattern

Reply:
We can still use the enums for assignment and array indexes and stuff right? I'd say it's not too big of a loss.

Reply:
I think what is at issue is a deeper question than just whether we use enums around the codebase as return or argument types.

I think the question is: what do we do when we have a style conflict with what ZRET OOT is doing.

I can tell you that ZRET MM doesn't always match OOT.

On the matter of the specific issue at hand... I see no reason the enum can't be used for our purposes (or even decomp, because you can use it only when matching, but this is a different discussion)

So before we make a decision, I'd like us to discuss and make a decision on if we even want to change documentation from ZRET OOT.

Reply:
From my experience, the only time I've ever had issues with using enum types as parameters or return types is when you have a need to use a value of a different enum type instead. But this seems like a pretty narrow use case, the only time I've ever really run into it is with the get-item-rework PR I've been working on, where I needed some functions that were only expecting `GetItemID` values to also accept/return `RandomizerGet` values. Even then if you're in that situation you'd probably prefer that the function wasn't tied to a specific enum by it's return/parameter types.